src/csharp/Microsoft.Spark/Network/DefaultSocketWrapper.cs

```diff
@@ -24,12 +25,12 @@ internal sealed class DefaultSocketWrapper : ISocketWrapper
         /// Default constructor that creates a new instance of DefaultSocket class which represents
         /// a traditional socket (System.Net.Socket.Socket).
         /// 
-        /// This socket is bound to Loopback with port 0.
+        /// This socket is bound to provided IP address with port 0.
         /// </summary>
-        public DefaultSocketWrapper() :
+        public DefaultSocketWrapper(IPAddress ipAddress) :
             this(new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
```

Comment:
For the IPv6 loopback, this doesn't work. It needs to be AddressFamily.InterNetworkV6 I think.

Reply:
I think we can fixing this by changing the code to:
this(new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp))

So that the address family of the socket matches the address family of the parsed IP address